x-pack/plugins/security_solution/scripts/endpoint/common/fleet_server/fleet_server_services.ts

```diff
@@ -0,0 +1,673 @@
+          body: newFleetHostEntry,
+        })
+        .catch(catchAxiosErrorFormatAndThrow)
+        .catch((error: FormattedAxiosError) => {
+          if (
+            error.response.status === 403 &&
+            ((error.response?.data?.message as string) ?? '').includes('disabled')
+          ) {
+            log.error(`Attempt to update fleet server host URL in fleet failed with [403: ${
+              error.response.data.message
+            }].
+
+  ${chalk.red('Are you running this utility against a Serverless project?')}
+  If so, the following entry should be added to your local
+  'config/serverless.[project_type].dev.yml' (ex. 'serverless.security.dev.yml'):
+
+  ${chalk.bold(chalk.cyan('xpack.fleet.internal.fleetServerStandalone: false'))}
+
+  `);
+          }
+
+          throw error;
+        })
+        .then((response) => response.data);
+
+      log.verbose(item);
+      log.info(`Fleet settings updated with fleet host URL successful`);
+      return item;
+    } catch (error) {
+      log.error(dump(error));
+      throw error;
+    }
+  });
+};
+
+const updateFleetElasticsearchOutputHostNames = async (
+  kbnClient: KbnClient,
+  log: ToolingLog
+): Promise<void> => {
+  log.info('Checking if Fleet output for Elasticsearch needs to be updated');
+
+  return log.indent(4, async () => {
+    try {
+      const localhostRealIp = getLocalhostRealIp();
+      const fleetOutputs = await fetchFleetOutputs(kbnClient);
+
+      // make sure that all ES hostnames are using localhost real IP
+      for (const { id, ...output } of fleetOutputs.items) {
+        if (output.type === 'elasticsearch') {
+          if (output.hosts) {
+            let needsUpdating = false;
+            const updatedHosts: Output['hosts'] = [];
+
+            for (const host of output.hosts) {
+              const hostURL = new URL(host);
+
+              if (isLocalhost(hostURL.hostname)) {
+                needsUpdating = true;
+                hostURL.hostname = localhostRealIp;
+                updatedHosts.push(hostURL.toString());
+
+                log.verbose(
+                  `Fleet Settings for Elasticsearch Output [Name: ${
+                    output.name
+                  } (id: ${id})]: Host [${host}] updated to [${hostURL.toString()}]`
+                );
+              } else {
+                updatedHosts.push(host);
+              }
+            }
+
+            if (needsUpdating) {
+              const update: PutOutputRequest['body'] = {
+                ...(output as PutOutputRequest['body']), // cast needed to quite TS - looks like the types for Output in fleet differ a bit between create/update
+                hosts: updatedHosts,
+              };
+
+              log.info(`Updating Fleet Settings for Output [${output.name} (${id})]`);
+
+              await kbnClient
+                .request<GetOneOutputResponse>({
+                  method: 'PUT',
+                  headers: { 'elastic-api-version': '2023-10-31' },
```

Comment:
I see you use `API_VERSIONS.public.v1` is it possible to use it also here?

Reply:
That `const` never made sense to me and I think will lead to issues. I was never really sure why a const was used. Version number will diverge at the api level (is my understanding) not the release level across all APIs